lib/initializer.rb

```diff
@@ -0,0 +1,35 @@
+class Initializer
+  class VoidBoardSizeError < StandardError
+    # attr_reader :action
+    # def initialize(message)
+    #   # call parent's constructor to set the message
+    #   super(message)
+    #   # store action in instance variable
+    #   # @action = action
+    # end
+  end
+
+  attr_accessor :input, :output, :board
+
+  def initialize(input=STDIN, output=STDOUT)
+    @input = input
+    @output = output
+    @board = Board.new(get_board_size)
+  end
+
+  def get_board_size
```

Comment:
I suggest looking into how to test error handling.

Reply:
I re-wrote this code so that a new function `create_board` initializes Board with the result of calling `get_board_size`. Now @board = `create_board`

the test for this new code is below. When I run this test though it just hangs in the terminal

    it "raises error if board size is less than 3" do
        @input = StringIO.new
        @output = StringIO.new
        # 2 is an invalid board size that should raise the VoidBoardSizeError
        @input.string = "2"
        @display = Display.new(@input, @output)

        expect { Initializer.new(@display) }.to raise_error("Board size is invalid. Size must be a positive integer greater than 2. Try again...")
      end